File: Leetcode/Unique Paths/Unique Paths.java
class Solution {
    public int uniquePaths(int m, int n) {
        //Initialising dp array
        Integer[][] dp = new Integer[m][n];
        //If n=0,then dp value will be 1
        for(int i = 0; i<m;i++){
            dp[i][0] = 1;
        }
        //If m=0,then dp value will be 1
        for(int j= 0;j<n;j++){
            dp[0][j]=1;
        }
        //For all the other points, ans will be sum of up and left point ans
        for(int i = 1;i<m;i++){
            for(int j = 1;j<n;j++){
                dp[i][j] = dp[i-1][j]+dp[i][j-1];
            }
        }
        //Return the final ans
        return dp[m-1][n-1];
    }
}

File: Leetcode/Unique Paths/Unique paths.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {

public:
    int uniquePaths(int m, int n)
    {
         //create a dp table of size (mxn) and initialize all values to zero      
        vector<vector<long long>> dp(m, vector<long long>(n,0));
        //initialize the first row and column vlaues to 1  
         for(int i=0;i<m;++i)        
         { 
             for(int j=0; j<n ;++j) 
            {
                if(i==0 || j==0)
                {
                    //if we are at 1st row or 1st column so initialize to value 1
                    dp[i][j]=1;
                }
                else
                {
                    //else dp[i][j] = sum of left cell +above cell because we can arrive a cell at(i,j)
                  dp[i][j] = dp[i-1][j] + dp[i][j-1];   
                }
            }
         }
        // return the vlaue in the last cell  
        return dp[m-1][n-1];
    }
};